graylog2-rest-client/src/main/java/org/graylog2/restclient/lib/ApiClientImpl.java

```diff
@@ -519,14 +523,22 @@ private void ensureAuthentication() {
             }
         }
 
+        private class RequestAndFuture {
```

Comment:
Maybe call it `RequestContext` and also put the `Node` object inside this class. This would enable us to put it inside a `Set<RequestContext>` and also simplify the iteration over that set over the use of `Map<Node, RequestAndFuture>`. That would require a valid `hashCode()` implementation, though (which can be done by using `@AutoValue`).